src/test/scala/org/bitcoins/core/script/constant/ScriptNumberSpec.scala

```diff
@@ -0,0 +1,59 @@
+package org.bitcoins.core.script.constant
+
+import org.bitcoins.core.gen.NumberGenerator
+import org.bitcoins.core.util.BitcoinSLogger
+import org.scalacheck.{Prop, Properties}
+
+/**
+  * Created by tom on 7/5/16.
+  */
+class ScriptNumberSpec extends Properties("ScriptNumberSpec") with BitcoinSLogger {
+  property("Additive identity") =
+    Prop.forAll(NumberGenerator.scriptNumbers) { num : ScriptNumber =>
+      num + ScriptNumber.zero == num
+    }
+  property("Subtraction identity") =
+    Prop.forAll(NumberGenerator.scriptNumbers) { num : ScriptNumber =>
+      num - ScriptNumber.zero == num
+    }
+  property("Multiplicative identity") =
+    Prop.forAll(NumberGenerator.scriptNumbers) { num : ScriptNumber =>
+      num * ScriptNumber.one == num
+    }
+  property("< >=") =
+    Prop.forAll(NumberGenerator.scriptNumbers, NumberGenerator.scriptNumbers) { (num1 : ScriptNumber, num2 : ScriptNumber) =>
+      if (num1.underlying < num2.underlying) num1 < num2
+      else num1 >= num2
+    }
+  property("> <=") =
+    Prop.forAll(NumberGenerator.scriptNumbers, NumberGenerator.scriptNumbers) { (num1 : ScriptNumber, num2 : ScriptNumber) =>
+      if (num1.underlying > num2.underlying) num1 > num2
+      else num1 <= num2
+    }
+  property("== & !=") = {
+    Prop.forAll(NumberGenerator.scriptNumbers, NumberGenerator.scriptNumbers) { (num1 : ScriptNumber, num2 : ScriptNumber) =>
+      if (num1.underlying == num2.underlying) num1 == num2
+      else num1 != num2
+    }
+  }
+  property("add two script numbers") =
+    Prop.forAll(NumberGenerator.scriptNumbers, NumberGenerator.scriptNumbers) { (num1 : ScriptNumber, num2: ScriptNumber) =>
+      num1 + num2 == ScriptNumber(num1.underlying + num2.underlying)
+    }
+  property("subtract a script number from another script number") =
+    Prop.forAll(NumberGenerator.scriptNumbers, NumberGenerator.scriptNumbers) { (num1 : ScriptNumber, num2: ScriptNumber) =>
+      num1 - num2 == ScriptNumber(num1.underlying - num2.underlying)
+    }
+  property("multiply two script numbers") =
+    Prop.forAll(NumberGenerator.scriptNumbers, NumberGenerator.scriptNumbers) { (num1 : ScriptNumber, num2: ScriptNumber) =>
+      num1 * num2 == ScriptNumber(num1.underlying * num2.underlying)
+    }
+  property("multiply a script number by zero should return zero") =
+    Prop.forAll(NumberGenerator.scriptNumbers) { (num1 : ScriptNumber) =>
+      num1 * ScriptNumber.zero == ScriptNumber.zero
+    }
+
```

Comment:
Can we think of a clever way to test the property of script numbers that says `We can perform an arithmetic operation on a ScriptNumber the results in a number larger than 4 bytes, but we cannot perform ANOTHER arithmetic operation on that result`?

For instance I think 
```scala
//should be valid
val result = ScriptNumber(Int.MaxValue) + ScriptNumber(Int.MaxValue)
//invalid because result is larger than 4 bytes
result + ScriptNumber(Int.MaxValue)
```